controllers/selfnoderemediationconfig_controller.go

```diff
@@ -243,3 +255,32 @@ func (r *SelfNodeRemediationConfigReconciler) convertTolerationsToUnstructed(tol
 	}
 	return convertedTolerations, nil
 }
+
+func (r *SelfNodeRemediationConfigReconciler) removeOldDsOnUpdateOperator(ctx context.Context) error {
+	ds := &v1.DaemonSet{}
+	key := types.NamespacedName{
+		Namespace: r.Namespace,
+		Name:      dsName,
+	}
+
+	if err := r.Client.Get(ctx, key, ds); err == nil {
+		if lastChangedActualVal, _ := ds.Annotations[lastChangedAnnotationKey]; lastChangedActualVal == lastChangeAnnotationVal {
```

Comment:
NIT: `lastChangedActualVal` and `lastChangeAnnotationVal` are a little bit similar. Can we use `Expected` rather than `Actual`?

Reply:
I'm not sure `Expected` fits this use case, since I don't expect old value to equal it, however I do agree that using `Actual` usually makes sense when coupled with `Expected` .
How about replacing `lastChangedActualVal` with `lastChangedFoundVal` ?

Reply:
SGTM